File: docs/scripting/samples/csharp-scripts/examples/Cancel_Plate.cs
/**
 * Cancel Plate
 * 
 * Cancels remaining protocol steps for specific plates based on conditions
 */

#region Header
// ** Copyright HighRes Biosolutions, Inc. 2016 **
// 
//    File:      CancelPlateScript.cs 
//    Project:   ScriptLibrary
//    Solution:  CellarioWPF
#endregion

using System.Linq;
using HRB.Cellario.Scripting.API;

// NOTE: When copying, change namespace to avoid run-time name clash with this example
namespace HRB.Cellario.Scripting.ScriptLibrary
{
    /// <summary>
    /// Example script for cancelling all further operations for a sample
    /// </summary>
    public class CancelPlateScript : AbstractScript
    {
        public override void Execute(IScriptingApi api)
        {
            var current = api.CurrentPlate;
            if (current.Name.EndsWith("2"))
            {
                // this example cancels all remaing steps for the current sample if the sample name ends in 2
                var cancelled = current.Cancel();
            }
        }
    }
}

File: docs/scripting/samples/python-scripts/python-examples/Cancel_Plate.py
"""
Cancel Plate

Cancels remaining protocol steps for specific plates based on conditions
"""

#Cellario modules
from HRB.Cellario.Scripting import *
from HRB.Cellario.Scripting.API import *



# Method called during Cellario protocol execution when a sample arrives at the scripting step.
# <remarks>Executes synchronously with the run scheduler. Device operation results not available
# until <see cref="ReleaseResources"/> method is called.</remarks>
# <Object name="api">Access to properties and methods for interacting with the Cellario run-time scheduler, 
# samples, resources and devices operations.</param>
def Execute(api : PythonScriptingApi):
    current = api.CurrentPlate
    if (current.Name.endswith("2")):
        #this example cancels all remaing steps for the current sample if the sample name ends in 2
        current.Cancel()
        api.Messaging.Notify("Cancel Plate Modal", "This sample: " + api.CurrentPlate.Name + " cancels all remaing steps for the current sample")